eth/core/precompile/factories.go

```diff
@@ -92,61 +97,99 @@ func (sf *StatefulFactory) Build(
 ) (vm.PrecompileContainer, error) {
 	sci, ok := utils.GetAs[StatefulImpl](rp)
 	if !ok {
-		return nil, errors.Wrap(ErrWrongContainerFactory, statefulContainerName)
+		return nil, errorslib.Wrap(ErrWrongContainerFactory, statefulContainerName)
 	}
 
 	// attach the precompile plugin to the stateful contract
 	sci.SetPlugin(p)
 
-	var err error
-
 	// add precompile methods to stateful container, if any exist
 	var idsToMethods map[string]*Method
-	if precompileMethods := sci.PrecompileMethods(); precompileMethods != nil {
-		idsToMethods, err = sf.buildIdsToMethods(precompileMethods, sci.ABIMethods())
-		if err != nil {
-			return nil, err
-		}
+	idsToMethods, err := BuildIdsToMethods(sci.ABIMethods(), reflect.ValueOf(sci))
+	if err != nil {
+		return nil, err
 	}
 
 	return NewStateful(rp, idsToMethods), nil
 }
 
-// buildIdsToMethods builds the stateful precompile container for the given `precompileMethods`
-// and `abiMethods`. This function will return an error if every method in `abiMethods` does not
-// have a valid, corresponding `Method`.
-func (sf *StatefulFactory) buildIdsToMethods(
-	precompileMethods Methods,
-	abiMethods map[string]abi.Method,
-) (map[string]*Method, error) {
-	// validate precompile methods
-	for _, pm := range precompileMethods {
-		if err := pm.ValidateBasic(); err != nil {
-			return nil, err
-		}
-	}
-
-	// match every ABI method to corresponding precompile method
+// This function matches each Go implementation of the Precompile
+// to the ABI's respective function.
+// It first searches for the ABI function in the Go implementation. If no find, then panic.
+// It then performs some basic validation on the implemented function
+// Then, the implemented function's arguments are checked against the ABI's arguments' types.
```

Comment:
style nitpick: write the godoc comments as a paragraph. so extend each of the lines to 100 characters